kratos/modeler/clean_up_problematic_triangles_modeler.h

```diff
@@ -0,0 +1,337 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "modeler/modeler.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class CleanUpProblematicTrianglesModeler
+ * @ingroup KratosCore
+ * @brief This class is responsible for cleaning up problematic geometries in a mesh, specifically null area geometries.
+ * @details The class identifies and removes degenerated triangles where two nodes are in the same position. It ensures that the mesh remains water-tight by reconnecting the sides and renumbering the nodes and entities.
+ * @note This is a solution to clean up problematic geometries in the mesh that for example can be found in OBJ files.
+ * @see Modeler
+ * @author Vicente Mataix Ferrandiz
+ */
+class KRATOS_API(KRATOS_CORE) CleanUpProblematicTrianglesModeler
+    : public Modeler
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Definition of the base type
+    using BaseType = Modeler;
+
+    /// Pointer definition of CleanUpProblematicTrianglesModeler
+    KRATOS_CLASS_POINTER_DEFINITION(CleanUpProblematicTrianglesModeler);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Constructor with Model
+     * @param Model The model to be used.
+     * @param ModelerParameters The parameters for the modeler.
+     */
+    CleanUpProblematicTrianglesModeler(
+        Model& rModel,
+        Parameters ModelerParameters = Parameters())
+        : BaseType(rModel, ModelerParameters),
+          mrModelPart(rModel.GetModelPart(ModelerParameters["model_part_name"].GetString()))
+    {
+    }
+
+    /**
+     * @brief Constructor with ModelPart
+     * @param rSourceModelPart The model part to be used.
+     * @param ModelerParameters The parameters for the modeler.
+     */
+    CleanUpProblematicTrianglesModeler(
+        ModelPart& rSourceModelPart,
+        Parameters ModelerParameters = Parameters()
+        ) : BaseType(rSourceModelPart.GetModel(), ModelerParameters),
+            mrModelPart(rSourceModelPart)
+    {
+        mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+    }
```

Comment:
I'd avoid any model-parameter constructor in here. Note that the `Modeler` object is rather new (or at least its new generation) so I'd promote a clean and standarized usage.

Reply:
What is the standard?, model/parameters?

Reply:
Yes.

Reply:
Okay